### PR TITLE
[doc] Add pulsar-client-go oauth2 authentication document

### PR DESCRIPTION
**Pull request by zymap**
---

*Motivation*

Add pulsar-client-go oauth2 authentication usage document.

### CODE REVIEW

**Comment by wolfstudy** [comment]
@Jennifer88huang PTAL thanks

File: site2/docs/security-oauth2.md [hunk 1]
```diff
@@ -113,3 +113,23 @@ config.setAuth(pulsar::AuthOauth2::create(params));
 
 pulsar::Client client("pulsar://broker.example.com:6650/", config);
 ```
+
+### Go client
+
+For enabling the oauth2 authentication in Go client. You need to configure the oauth2 authentication.
```
Comment (Huanli-Meng) [inline site2/docs/security-oauth2.md:119]:
```suggestion
To enable Oauth2 authentication in Go client, you need to configure Oauth2 authentication.
```

File: site2/docs/security-oauth2.md [hunk 2]
```diff
@@ -113,3 +113,23 @@ config.setAuth(pulsar::AuthOauth2::create(params));
...
+
+Here's an example:
```
Comment (Huanli-Meng) [inline site2/docs/security-oauth2.md:121]:
```suggestion
This example shows how to configure Oauth2 authentication in Go client. 
```

File: site2/docs/client-libraries-go.md [hunk 3]
```diff
@@ -658,3 +658,23 @@ opts := pulsar.ClientOptions{
     Authentication: NewAuthenticationTLS("my-cert.pem", "my-key.pem"),
 }
 ```
+
+## OAuth2 authentication
+
+In order to use [OAuth2 authentication](security-oauth2.md), you'll need to configure your client to do so:
```
Comment (Huanli-Meng) [inline site2/docs/client-libraries-go.md:664]:
```suggestion
To use [OAuth2 authentication](security-oauth2.md), you'll need to configure your client to perform the following operations.
```

File: site2/docs/client-libraries-go.md [hunk 4]
```diff
@@ -658,3 +658,23 @@ opts := pulsar.ClientOptions{
...
+
+Here's an example:
```
Comment (Huanli-Meng) [inline site2/docs/client-libraries-go.md:666]:
```suggestion
This example shows how to configure Oauth2 authentication.
```

**Pull request merged**